<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>

<head>
<meta http-equiv="content-type" content="text/html; charset=utf-8">
<meta http-equiv="content-language" content="en-us">
<meta name="description" content="PhyloSort is a Java tool to sort phylogenetic trees by searching for user-specified subtrees that contain a monophyletic group of interest defined by operational taxonomic units.">
<meta name="keywords" content="phylosort, phylogenetic, tree, monophyletic, operational taxonomic unit, otu, lowest common ancestor, newick, parser, java, open source, api, linux, windows, macos, reroot, outgroup, monophyly, topology, taxa, taxon, gene, gene family, tree cluster">
<link type="text/css" rel="stylesheet" href="style.css">
<link rel="shortcut icon" href="/favicon.ico" type="image/x-icon">
<meta name="verify-v1" content="NrAXZrjpp+Hw3coQhNpOlDqkU2tDEi7FgwIRf3avKWs=">
<meta name="author" content="Ahmed Moustafa">
<title>PhyloSort</title>
</head>

<body>

<div id="container">

<div id="header">
<h1>PhyloSort</h1>
<p>A Java tool to sort phylogenetic trees by searching for user-specified subtrees that contain a <a href="http://en.wikipedia.org/wiki/Monophyletic">monophyletic</a> group of interest defined by operational taxonomic units (OTUs).</p>
</div>

<div id="menu">
&raquo;<a href="http://phylosort.sourceforge.net/">Homepage</a><br>
&raquo;<a href="http://sourceforge.net/project/showfiles.php?group_id=196947">Download</a><br>
<small>(<a href="http://sourceforge.net/project/showfiles.php?group_id=196947&amp;package_id=232914&amp;release_id=528477">Current release 1.3</a>)</small><br>
<a href="http://downloads.sourceforge.net/phylosort/phylosort_windows_1_3.exe"><img src="platform_windows.png" width="40" height="40" alt="Windows" title="Windows"></a>
<a href="http://downloads.sourceforge.net/phylosort/phylosort_linux_1_3.rpm"><img src="platform_linux.png" width="40" height="40" alt="Linux" title="Linux"></a>
<br>
<a href="http://downloads.sourceforge.net/phylosort/phylosort_macos_1_3.dmg"><img src="platform_mac_os_x.png" width="40" height="40" alt="Mac OS X" title="Mac OS X"></a>
<a href="http://downloads.sourceforge.net/phylosort/phylosort_source_1_3.zip"><img src="platform_java.png" width="40" height="40" alt="Source code" title="Source code"></a>
<br>
&raquo;<a href="http://sourceforge.net/projects/phylosort/">SF Project</a><br>
&raquo;<a href="http://phylosort.cvs.sourceforge.net/phylosort/">CVS Repository</a><br>
&raquo;<a href="api/">API Documentation</a><br>
&raquo;<a href="http://www.jars.com/science/biology/resource.php/37243">JARS Rated Top 25%</a><br>
&raquo;<a href="related.html">Related Publications</a><br>
</div>

<div id="content">

<h2>Features</h2>
<ul>
<li>Searching for monophyletic relationship among groups of taxa.</li>
<li>Filtering by <a href="http://www.icp.ucl.ac.be/~opperd/private/bootstrap.html">bootstrap</a> support values associated with the monophyletic clades.</li>
<li>Filtering by tree complexity (number of taxa in a tree).</li>
<li>Filtering by family complexity (number of genes per taxon in a tree).</li>
<li>Clustering trees (genes) into tree clusters (gene families).</li>
<li>Grouping OTUs using a taxonomy reference tree.</li>
<li>Clean and reusable implementation of common procedures to manipulate tree data structure such as Newick parsing, tree traversal and rerooting.</li>
<li>Visualizing tree topology using <a href="http://www.phylosoft.org/atv/">A Tree Viewer (ATV)</a>.</li>
</ul>

<div align="center">
<a href="screenshot1.png"><img class="screenshot" src="screenshot1_small.png" alt="" width="200" height="150" title="Screenshot #1: main window"></a>
<a href="screenshot2.png"><img class="screenshot" src="screenshot2_small.png" alt="" width="95" height="150" title="Screenshot #2: reference tree window"></a>
<a href="screenshot3.png"><img class="screenshot" src="screenshot3_small.png" alt="" width="166" height="150" title="Screenshot #3: phylogenetic tree window"></a>
</div>

<h2>Usage</h2>
<h3>Input</h3>

<ul>
<li>Select an input (readable) folder for the input trees. Once the input folder is selected, all trees will be scanned to retrieve the set of all taxa in all trees. These taxa will show in the list of available taxa on the left hand side of the main window.</li>
<li>Select an output (writeable) folder for the matching trees.</li>
<li>Define query taxa group(s) by moving taxa from the list of available taxa to the query taxa on the right hand side of the main window. Groups can be added and removed using the plus (+) and minus (&minus;) buttons, respectively.</li>
</ul>

<h3>Settings</h3>
<h4>Mode</h4>
<p>PhyloSort offers two search modes, <em>inclusive</em> (default) and <em>exclusive</em>:</p>
<ul>
<li>In the exclusive mode, any taxa from the query group(s) exist in the tree, will have to be located within the monophyletic clade in order to qualify that clade and consider the tree as a match tree.</li>
<li>In the inclusive mode, the restriction of the exclusive mode is relaxed such that a &quot;qualifying&quot; monophyletic clade is accepted regardless whether there are additional taxa from the query taxa located somewhere else in the tree outside that clade.</li>
</ul>

<h4>Root</h4>
<p>PhyloSort searches for an <a href="http://en.wikipedia.org/wiki/Outgroup">outgroup</a> and reroots the tree on the identified outgroup. If an outgroup could not be found (i.e. all taxa in the tree are subset of the query taxa), the tree is searched without rerooting.</p>

<h4>Filtering</h4>
<p>The following is a list of filtering parameters that could be used to adjust the matching output trees (any of these filters can be turned off by setting its value to a negative number).</p>

<ul>
<li>Minimum number of taxa: the least number of taxa must be contained in a tree. (integer number)</li>
<li>Maximum number of taxa: the largest number of taxa a tree can contain. (integer number)</li>
<li>Minimum <a href="http://www.icp.ucl.ac.be/~opperd/private/bootstrap.html">bootstrap</a> support: the minimum <a href="http://www.icp.ucl.ac.be/~opperd/private/bootstrap.html">bootstrap</a> support value on the <em>monophyletic</em> clade. (decimal number)</li>
<li>Maximum average number of genes (or copies) per taxon. (decimal number)</li>
</ul>

<p>All the settings and filters can be loaded from a configuration file, which may include an item or more to configure. The configuration file is a key-value-pair-based text file. In the GUI interface, a configuration file can be loaded via the menu "File" &rarr; "Load configuration" (or shortcut <code>Ctrl+O</code>). In the command line interface, it can be loaded through the system environment <code>phylosort.config</code> (e.g. <code>-Dphylosort.config=filename</code>). The following list shows all configurable properties and their possible values.</p>

<ul>
<li><code>phylosort.root.outgroup</code> : <code>yes</code> (default) or <code>no</code></li>
<li><code>phylosort.mode</code> : <code>inclusive</code> (default) or <code>exclusive</code></li>
<li><code>phylosort.minimum.number.taxa</code> : integer number</li>
<li><code>phylosort.maximum.number.taxa</code> : integer number</li>
<li><code>phylosort.minimum.bootstrap.support</code> : decimal number</li>
<li><code>phylosort.maximum.average.number.copies</code> : decimal number</li>
<li><code>phylosort.on.match.action</code> : <code>copy</code>, <code>move</code> or <code>count</code> (default)</li>
<li><code>phylosort.pattern</code> : <a href="http://en.wikipedia.org/wiki/regular_expression" title="regular expression">regular expression</a></li>
</ul>

<h3>Output</h3>
<p>Based on the selection criteria and the <code>phylosort.on.match.action</code>, trees with matching monophyletic clades are counted, copied or moved from the input folder to the output folder.</p>

<h3>Command line</h3>
<h4>Command line arguments</h4>
<div class="syntax">java [-Dphylosort.config=&lt;configfile&gt;] -jar phylosort.jar &lt;infolder&gt; &lt;outfolder&gt; &lt;taxafile&gt;</div>


<h4>Graphical user interface</h4>
<div class="syntax">java [-Dphylosort.config=&lt;configfile&gt;] -jar phylosort.jar</div>



<h3>Prerequisites</h3>
<p>PhyloSort requires that <a href="http://en.wikipedia.org/wiki/Java_Runtime_Environment">Java Runtime Environment</a> (JRE) version 1.5 or later is installed. JRE can be downloaded at <a href="http://www.java.com/">java.com</a>.</p>

<h2>Licenses</h2>
<p>The source code is licensed under <a href="http://www.gnu.org/licenses/gpl.txt">The GNU General Public License (GPL)</a>.</p>
<p>If you are using PhyloSort in a published work or product, please cite: <a href="http://www.biomedcentral.com/1471-2148/8/6">Ahmed Moustafa and Debashish Bhattacharya. PhyloSort: A user-friendly phylogenetic sorting tool and its application to estimating the cyanobacterial contribution to the nuclear genome of <em>Chlamydomonas</em>. BMC Evol. Biol. 8:6</a> (open access) (Qualified as <a href="http://www.biomedcentral.com/info/about/mostviewed#highlyaccessed">highly accessed</a>).</p>

<h2>Acknowledgments</h2>
<ul>
<li><a href="http://www.nih.gov/">National Institutes of Health (NIH)</a> for supporting our research.</li>
<li><a href="http://www.ej-technologies.com/index.html">ej-technologies</a> for providing a license for <a href="http://www.ej-technologies.com/products/install4j/overview.html">install4j</a> to generate the multi-platform installers.</li>
</ul>

</div>

<div id="footer">
<a href="http://sourceforge.net/"><img src="http://sourceforge.net/sflogo.php?group_id=196947&amp;type=10" alt="SourceForge.Net" title="SourceForge.Net" width="80" height="15"></a> <a href="http://www.gnu.org/licenses/gpl.txt"><img src="gpl.png" alt="GNU General Public License" title="GNU General Public License" width="80" height="15"></a> <a href="http://www.gnu.org/licenses/fdl.txt"><img src="gfdl.png" alt="GNU Free Documentation License" title="GNU Free Documentation License" width="80" height="15"></a> <a href="http://www.anybrowser.org/campaign/"><img src="anybrowser.png" alt="Any Browser" title="Viewable With Any Browser" width="80" height="15"></a> <a href="http://validator.w3.org/check?uri=referer"><img src="validhtml401.png" alt="Valid HTML 4.01 Transitional" title="Valid HTML 4.01 Transitional" width="80" height="15"></a> <a href="http://jigsaw.w3.org/css-validator/check/referer"><img src="csstableless.png" alt="Valid CSS!" title="Valid CSS!" width="80" height="15"></a>
<br>
Written and maintained by <a href="http://ahmedmoustafa.com">Ahmed Moustafa</a> (<em><a href="mailto:ahmed@users.sf.net">ahmed@users.sf.net</a></em>) and <a href="http://dblab.rutgers.edu">Debashish Bhattacharya</a> (<em><a href="mailto:bhattacharya@aesop.rutgers.edu">bhattacharya@aesop.rutgers.edu</a></em>)
<br>
Last modified $Date: 2009/09/23 13:34:02 $


<!-- Start of StatCounter Code -->

<script type="text/javascript" language="javascript">
<!-- 
var sc_project=2574704; 
var sc_invisible=0; 
var sc_partition=25; 
var sc_security="f0ff4776"; 

//-->
</script>

<script type="text/javascript" language="javascript" src="http://www.statcounter.com/counter/counter.js"></script>

<noscript>
<img src="http://c26.statcounter.com/counter.php?sc_project=2574704&amp;java=0&amp;security=f0ff4776&amp;invisible=0" alt="" border="0" width="0" height="0">
</noscript>

<!-- End of StatCounter Code -->

</div>

</div>

<!-- Start of Google Analytics -->

<script src="http://www.google-analytics.com/urchin.js" type="text/javascript">
</script>
<script type="text/javascript">
_uacct = "UA-91815-3";
urchinTracker();
</script>

<!-- End of Google Analytics -->

</body>
</html>
